lib/spack/spack/package.py

```diff
@@ -1244,7 +1244,7 @@ def do_fetch(self, mirror_only=False):
                                  self.spec.format('{name}{@version}'), ck_msg)
 
         self.stage.create()
-        self.stage.fetch(mirror_only)
+        self.stage.fetch(mirror_only, self.manual_download)
```

Comment:
Just to double check. There's another call to `Stage.fetch` here:

https://github.com/spack/spack/blob/20d37afafa3be3f9b6c3388e22f813261d4c3fec/lib/spack/spack/patch.py#L192-L194

but I don't think that will ever cause inconsistent behavior (i.e. I don't think we'll ever be able to call the same or equivalent stage object with `manual_download=True` here and `manual_download=False` at the lines above). Wondering if you agree with that.

Reply:
Thanks for pointing me to that call.  The comment is outdated as `stage` is no longer an argument for `UrllPatch` since it is created on demand.  Curious that a `mirror_only` option does not appear to be associated with patch fetching.  Should it be?

I am hesitant to say/think a mismatch could *never* be an issue though I would think it unlikely that patches for a code that is required to be manually downloaded would be made available from a site that does not require manual download.  From a built-in package perspective, it appears the patches are downloaded with the source.  For example, `Molcas` is licensed software that has a patch but the url is an archive file so the patch must be co-located in the expanded source directory.  The `fsl`, `Gapfiller`, `Signalp`, `Targetp`, and `Tmhmm` packages similarly specify archive file urls and provide `patch` methods.

On a related note, do you know why patching doesn't support `mirror_only` fetching?